Looms.PoS/Looms.PoS.Application/Features/OrderItem/Commands/UpdateOrderItem/UpdateOrderItemRequestValidator.cs

```diff
@@ -10,38 +11,98 @@ public class UpdateOrderItemRequestValidator : AbstractValidator<UpdateOrderItem
+            .CustomAsync(async (productId, context, _) =>
             {
-                var discount = await discountsRepository.GetAsync(Guid.Parse(discountId));
+                var productDao = await productsRepository.GetAsyncByIdAndBusinessId(
+                    Guid.Parse(productId!),
+                    Guid.Parse((string)context.RootContextData[HeaderConstants.BusinessIdHeader])
+                );
 
-                if (discount.Target == DiscountTarget.Order)
+                if (productDao.Quantity < context.InstanceToValidate.Quantity)
                 {
-                    context.AddFailure("Discount cannot be applied to order item.");
+                    context.AddFailure("Product quantity is too low.");
                 }
             })
-            .When(x => x.DiscountId is not null);
+            .When(x => x.ProductId is not null);
 
-        RuleFor(x => x.VariationId!)
+        RuleFor(x => x.ProductVariationId)
+            .Null()
+            .When(x => x.ProductId is null)
+            .WithMessage("ProductVariation is only allowed with Product.");
+
+        RuleFor(x => x.ProductVariationId)
             .MustBeValidGuid()
-            .CustomAsync(async (variationId, context, cancellationToken) =>
+            .CustomAsync(async (variationId, context, _) =>
             {
-                var variation = await variationsRepository.GetAsync(Guid.Parse(variationId));
+                var variation =
+                    await variationRepository.GetAsyncByIdAndProductId(
+                        Guid.Parse(variationId!),
+                        Guid.Parse(context.InstanceToValidate.ProductId!)
+                    );
 
-                if (variation.Quantity < context.InstanceToValidate.Quantity)
+                if (variation!.Quantity < context.InstanceToValidate.Quantity)
                 {
-                    context.AddFailure("Product variation quantity is too low.");
+                    context.AddFailure("Variation quantity is too low.");
                 }
             })
-            .When(x => x.VariationId is not null);
+            .When(x => x.ProductVariationId is not null);
+
+        RuleFor(x => x.DiscountId)
+            .MustBeValidGuid()
+            .CustomAsync(async (discountId, context, _) =>
+            {
+                var discount = await discountsRepository.GetAsyncByIdAndBusinessId(
+                    Guid.Parse(discountId!),
+                    Guid.Parse((string)context.RootContextData[HeaderConstants.BusinessIdHeader])
+                );
+
+                if (discount.Target == DiscountTarget.Order)
+                {
+                    context.AddFailure("Discount cannot be applied to order item.");
+
+                    return;
+                }
+
+                if (context.InstanceToValidate.ProductId is not null
+                 && discount.ProductId != Guid.Parse(context.InstanceToValidate.ProductId))
```

Comment:
will throw if product/service is invalid guid and discount is valid non order type